cmd/soci/commands/ztoc/list.go

```diff
@@ -30,25 +34,69 @@ var listCommand = cli.Command{
 	Description: "list ztocs",
 	Flags: []cli.Flag{
 		cli.StringFlag{
-			Name:  "digest",
+			Name:  "ztoc-digest",
 			Usage: "filter ztocs by digest",
 		},
+		cli.StringFlag{
+			Name:  "image-ref",
+			Usage: "filter ztocs to those that are associated with a specific image",
+		},
 	},
 	Action: func(cliContext *cli.Context) error {
 		db, err := soci.NewDB()
 		if err != nil {
 			return err
 		}
-		digest := cliContext.String("digest")
+		ztocDgst := cliContext.String("ztoc-digest")
+		imgRef := cliContext.String("image-ref")
 
 		var artifacts []*soci.ArtifactEntry
-		db.Walk(func(ae *soci.ArtifactEntry) error {
-			if ae.Type == soci.ArtifactEntryTypeLayer &&
-				(digest == "" || ae.Digest == digest) {
-				artifacts = append(artifacts, ae)
+		if imgRef == "" {
+			db.Walk(func(ae *soci.ArtifactEntry) error {
+				if ae.Type == soci.ArtifactEntryTypeLayer && (ztocDgst == "" || ae.Digest == ztocDgst) {
+					artifacts = append(artifacts, ae)
+				}
+				return nil
+			})
+		} else {
+			client, ctx, cancel, err := commands.NewClient(cliContext)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			is := client.ImageService()
+			img, err := is.Get(ctx, imgRef)
+			if err != nil {
+				return err
+			}
+			platform, err := images.Platforms(ctx, client.ContentStore(), img.Target)
+			if err != nil {
+				return err
 			}
-			return nil
-		})
+			var layers []ocispec.Descriptor
+			for _, p := range platform {
+				manifest, err := images.Manifest(ctx, client.ContentStore(), img.Target, platforms.OnlyStrict(p))
+				if err == nil {
+					layers = append(layers, manifest.Layers...)
+					break
```

Comment:
Why do we break here? What if the user has created indices for multiple platforms?

Reply:
Here, we only need to find the image layers.

Reply:
Image layers depend on the platform. If you create an index for 2 platforms, this will only show the ztocs for 1 platform.

Reply:
> That is true only if the image have different layers for different platforms.

This is likely to be true for any layer that contains binaries.

> If we don't break here, the next `images.Manifest` call will error out as it walks through the image again trying to read all manifests again and that read will fail.

I'm not sure I understand this. I would expect `images.Manifest` to succeed for every platform that I have pulled.

Reply:
Just found out some image can have multiple platforms in their image index, but manifests for some platforms don't exist, which will result in a failed ``images.Manifest`` call for those platforms.  In this case, we will print a warning message and continue looking for manifests other other platforms. I've updated this in the latest revision.